### PR TITLE
Tests: Upgrade QUnit to 2.0

### PR DESCRIPTION
**Pull request by Arkni**
Ref #1681

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,17 +37,18 @@
...
-		"grunt-contrib-copy": "^0.5",
-		"grunt-contrib-jshint": "^0.11.3",
-		"grunt-contrib-qunit": "^0.4",
-		"grunt-contrib-uglify": "^0.4",
-		"grunt-contrib-watch": "^0.6",
-		"grunt-jscs": "^2.2",
-		"grunt-text-replace": "^0.3.11"
+		"commitplease": "2.3.1",
+		"grunt": "1.0.1",
+		"grunt-contrib-compress": "1.2.0",
```
Comment (staabm) [inline package.json]: Why did you add this hardcoded constraints?

**Comment by staabm** [comment]
I like what I see so far

File: package.json (hunk 1 above)
Comment (Arkni) [inline package.json]: I followed the convention used in most of `jQuery` projects (`QUnit`, `jquery`, `jquery-ui` ...).

Here is a quote from one of the core developpers of `jQuery`:
>Most of jQuery repos went in the opposite direction. Semver is an idea but reality is that from time to time a package breaks and then we have a problem. IMO it's better to just pin versions and update from time to time. Note that using non-latest version of a dependency isn't a bad thing; maybe they introduced a new feature we don't use, maybe a bug fix that doesn't affect us that much. It's good to not lock down to something ancient but it doesn't mean constantly living on the bleeding edge is an adequate answer either

But if you want, I can leverage all the dev-dependencies.

Comment (staabm) [inline package.json]: As those are only devDeps I can live with whatever you use. Just curious

**Comment by Arkni** [comment]
I'll update this PR as soon as `QUnit@2.0` is released. Currently, only the `rc1` is available and I did this change to make sure everything works a expected with the new major version.

**Comment by Arkni** [comment]
@staabm I think this is good to go.

**Comment by staabm** [comment]
@Arkni I will create a 1.15.1 release now and we can land this PR after that. could you rebase it then?

may I ask you to create a changelog from the current master?

**Comment by Arkni** [comment]
> I will create a 1.15.1 release now and we can land this PR after that. could you rebase it then?

Sure, I will do that.

> may I ask you to create a changelog from the current master?

Of course, just a few minutes.

**Comment by Arkni** [comment]
@staabm 
Here you go: https://gist.github.com/Arkni/8775e2f0755659b98d38ca05b96fa798#file-jquery-validate-1-15-1-md

Cleaned it up a little :)

**Comment by Arkni** [comment]
And PR rebased as requested :)

**Comment by staabm** [comment]
@Arkni thank you so much for your assistance. couldnt have build this release without your support!!

**Comment by Arkni** [comment]
Any time :)
